community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -104,9 +91,35 @@ def instance_properties(nodeset:object, model:str, placement_group:Optional[str]
 
     # Override with properties explicit specified in the nodeset
     props.update(nodeset.get("instance_properties") or {})
-    
+
     return props
 
+def update_reservation_props(reservation:Optional[object], props:object) -> None:
+    if not reservation:
+        return
+
+    props.reservationAffinity = {
+        "consumeReservationType": "SPECIFIC_RESERVATION",
+        "key": f"compute.{util.universe_domain()}/reservation-name",
+        "values": [reservation.bulk_insert_name],
+    }
+
+    if reservation.policies:
+        props.scheduling.onHostMaintenance = "TERMINATE"
+        props.resourcePolicies = reservation.policies
+        log.info(
+            f"reservation {reservation.bulk_insert_name} is being used with policies {props.resourcePolicies}"
+        )
+    else:
+        props.resourcePolicies = []
+        log.info(
+            f"reservation {reservation.bulk_insert_name} is being used without any policies"
+        )
+
+def use_future_reservation(props:object, nodeset:object) -> None:
+    if (fr := lookup().future_reservation(nodeset)) and fr.active and fr.specific:
```

Comment:
If feature reservation is not "active" we shouldn't even attempt to provision VM. Your code just silently omits `reservationAffinity`

Reply:
Moving this function in-line in callee